src/components/common/SocialMediaLink.tsx

```diff
@@ -1,24 +1,24 @@
-import Link from "next/link";
 import React from "react";
+import { IconType } from "react-icons";
 
 interface SocialMediaLinkProps {
   href: string;
-  icon: React.ComponentType;
+  icon: IconType;
+  className?: string;
 }
 
 const SocialMediaLink: React.FC<SocialMediaLinkProps> = ({
   href,
   icon: Icon,
+  className = "",
 }) => {
   return (
-    <Link
+    <a
       href={href}
-      target="_blank"
-      rel="noopener noreferrer"
-      className="text-gray-500 hover:text-white"
+      className="bg-gray-800 p-3 rounded-full hover:bg-gray-700 ${className}"
```

Comment:
your template doesnt work
it should be:
```suggestion
      className={`bg-gray-800 p-3 rounded-full hover:bg-gray-700 ${className}`}
```